test/metabase/api/permissions_test.clj

```diff
@@ -102,6 +104,33 @@
       (is (= "You don't have permissions to do that."
            (mt/user-http-request :rasta :get 403 (format "permissions/group/%d" (:id (perms-group/all-users)))))))))
 
+(deftest create-group-test
+  (testing "POST /permissions/group"
+    (testing "happy path"
+      (mt/with-model-cleanup [PermissionsGroup]
+        (mt/user-http-request :crowberto :post 200 "permissions/group" {:name "Test Group"})
+        (is (some? (t2/select PermissionsGroup :name "Test Group")))))
+
+   (testing "requires superuser"
+     (is (= "You don't have permissions to do that."
+          (mt/user-http-request :rasta :post 403 "permissions/group" {:name "Test Group"}))))
+
+   (testing "group name is required"
+     (is (= {:errors {:name "value must be a non-blank string."}}
+            (mt/user-http-request :crowberto :post 400 "permissions/group" {:name nil}))))))
+
+(deftest delete-group-test
+  (testing "DELETE /permissions/group/:id"
+    (testing "happy path"
+      (t2.with-temp/with-temp [PermissionsGroup {group-id :id} {:name "Test group"}]
+        (mt/user-http-request :crowberto :delete 204 (format "permissions/group/%d" group-id))
+        (is (= 0 (t2/count PermissionsGroup :name "Test group")))))
```

Comment:
Maybe I'm hyper, but I usually put an

```
(is (= 1 (t2/count PermissionsGroup :name "Test group"))
```
beforehand just to be absolutely sure that the test does what I think it does. Consider, for example, that `(t2/count PermissionsGroup :name "Test Group")` (with a typo: capital `G`) will always return 0.

Reply:
yea I do that as well but in this case the test is quite simple and the with-temp is right above it so I think it's fine.